x/or/rollup/keeper/msg_server.go

```diff
@@ -0,0 +1,218 @@
+}
+
+func (k msgServer) RegisterSequencer(goCtx context.Context, msg *types.MsgRegisterSequencer) (*types.MsgRegisterSequencerResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	if _, found := k.GetSequencer(ctx, msg.Creator); found {
+		return nil, types.ErrSequencerExists
+	}
+
+	rollup, found := k.GetRollup(ctx, msg.RollupName)
+	if found {
+		permissionedAddresses := rollup.PermissionedAddresses.Addresses
+		if len(permissionedAddresses) > 0 {
+			permissioned := false
+			for i := range permissionedAddresses {
+				if permissionedAddresses[i] == msg.Creator {
+					permissioned = true
+					break
+				}
+			}
+			if !permissioned {
+				return nil, types.ErrSequencerNotPermissioned
+			}
+		}
+	} else {
+		return nil, types.ErrNotExistRollupName
+	}
+
+	sequencersByRollup, found := k.GetSequencersByRollupName(ctx, msg.RollupName)
+	sequencer := types.Sequencer{
+		SequencerAddress: msg.Creator,
+		Pubkey:           msg.Pubkey,
+		RollupName:       msg.RollupName,
+	}
+
+	if found {
+		// TODO: Modify when rollup:sequencer=1:N
+		return nil, types.SequencersByRollupExists
+	}
+	sequencersByRollup.RollupName = msg.RollupName
+	sequencersByRollup.Sequencers = append(sequencersByRollup.Sequencers, sequencer)
+
+	k.SetSequencersByRollup(ctx, sequencersByRollup)
+
+	k.SetSequencer(ctx, sequencer)
+
+	depositMsg := types.NewMsgDeposit(msg.RollupName, msg.Creator, msg.Value)
+	_, err := k.Deposit(goCtx, depositMsg)
+	if err != nil {
+		return nil, err
+	}
+
+	return &types.MsgRegisterSequencerResponse{}, nil
+}
+
+func (k msgServer) RemoveSequencer(goCtx context.Context, msg *types.MsgRemoveSequencer) (*types.MsgRemoveSequencerResponse, error) {
+	panic("implement me")
+}
+
+func (k msgServer) Deposit(goCtx context.Context, msg *types.MsgDeposit) (*types.MsgDepositResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	_, found := k.GetRollup(ctx, msg.RollupName)
+	if !found {
+		return nil, types.ErrNotExistRollupName
+	}
+
+	sequencersInfo, found := k.GetSequencersByRollupName(ctx, msg.RollupName)
+	if !found {
+		return nil, types.ErrNotExistSequencer
+	}
+	sequencerContains := false
+	for _, v := range sequencersInfo.Sequencers {
+		if v.SequencerAddress == msg.SequencerAddress {
+			sequencerContains = true
+			break
+		}
+	}
+	if !sequencerContains {
+		return nil, types.ErrNotExistSequencer
+	}
+
+	addr, err := sdk.AccAddressFromBech32(msg.SequencerAddress)
+	if err != nil {
+		return nil, err
+	}
+	amount := sdk.NewCoins(msg.Value)
+	err = k.bankKeeper.SendCoinsFromAccountToModule(ctx, addr, types.ModuleName, amount)
+	if err != nil {
+		return nil, err
+	}
+	deposit := types.Deposit{
+		RollupName:       msg.RollupName,
+		SequencerAddress: msg.SequencerAddress,
+		Value:            msg.Value,
+	}
+
+	k.SetDeposit(ctx, deposit)
+
+	return &types.MsgDepositResponse{}, nil
+}
+
+func (k msgServer) Slash(ctx sdk.Context, rollupName string, sequencerAddress string, value sdk.Coin) error {
```

Comment:
I think this function is used by only inner modules. Would you like to move this function to sequencer.go?

Reply:
I want to implement only Set*** and Get*** in sequencer.go, so move to keeper.go.